File: apps/merchant-api/PaymentService/src/PaymentService.Application/Dtos/Orders/OrderWithCompleteDetailsDto.cs
namespace PaymentService.Application.Dtos.Orders;

public class OrderWithCompleteDetailsDto
{
    public int OrderNumber{ get; set; }
    public string OrderStatus{ get; set; } = string.Empty;
    public string PaymentStatus{ get; set; } = string.Empty;
    public DateOnly CreatedOrderDate{ get; set; }
    public double SubTotalPrice { get; set; }
    public int Discount { get; set; }
    public double TotalPrice { get; set; }
    public List<OrderItemWithCompletedDetailsDto> OrderItems { get; set; } = [];
}

File: apps/merchant-web/src/schemes/order/OrderWithCompleteDetailsDto.ts
import {OrderItemWithCompletedDetailsDto} from "@/schemes/order/OrderItemWithCompletedDetailsDto";

export class OrderWithCompleteDetailsDto {
  orderNumber: number;
  orderStatus: string;
  paymentStatus: string;
  createdOrderDate: string;
  subTotalPrice: number;
  discount: number;
  totalPrice: number;
  orderItems: OrderItemWithCompletedDetailsDto[];

  constructor(
    orderNumber: number,
    orderStatus: string,
    paymentStatus: string,
    createdOrderDate: string,
    subTotalPrice: number,
    discount: number,
    totalPrice: number,
    orderItems: OrderItemWithCompletedDetailsDto[]
  ) {
    this.orderNumber = orderNumber;
    this.orderStatus = orderStatus;
    this.paymentStatus = paymentStatus;
    this.createdOrderDate = createdOrderDate;
    this.subTotalPrice = subTotalPrice;
    this.discount = discount;
    this.totalPrice = totalPrice;
    this.orderItems = orderItems;
  }
}